api/spec/requests/spree/api/v2/storefront/cart_spec.rb

```diff
@@ -100,4 +100,103 @@
+
+        expect(response.status).to eq(404)
+        expect(json_response[:error]).to eq("Order doesn't exist")
+      end
+    end
+
+    context 'existing order' do
+      let!(:order) { create(:order, user: user) }
+
+      context 'without line items' do
+        let!(:line_item) { create(:line_item) }
+
+        it 'tries to remove an item and fails' do
+          headers = { 'Authorization' => "Bearer #{token.token}" }
+          delete '/api/v2/storefront/cart/remove_item', params: { line_item_id: line_item.id }, headers: headers
+
+          expect(response.status).to eq(404)
+        end
+      end
+
+      context 'containing line item' do
+        let!(:line_item) { create(:line_item, order: order) }
+
+        it 'removes line item from the cart' do
+          headers = { 'Authorization' => "Bearer #{token.token}" }
+          delete '/api/v2/storefront/cart/remove_item', params: { line_item_id: line_item.id }, headers: headers
+
+          expect(response.status).to eq(200)
+
+          expect(order.line_items.count).to eq(0)
+
+          expect(json_response['data']).to have_id(order.id.to_s)
+          expect(json_response['data']).to have_type('cart')
+          expect(json_response['data']).to have_attribute(:number).with_value(order.number)
+          expect(json_response['data']).to have_attribute(:state).with_value('cart')
+          expect(json_response['data']).to have_relationships(:user, :line_items, :variants)
+        end
+      end
+    end
+
+    # context guest_token
+    context 'as a guest' do
+      let(:guest_token) { 'guest_token' }
+
+      context 'without existing order' do
+        let!(:line_item) { create(:line_item) }
+
+        it 'returns error' do
+          headers = { 'Authorization' => "Bearer #{token.token}" }
+          delete '/api/v2/storefront/cart/remove_item', params: { line_item_id: line_item.id }, headers: headers
+
+          expect(response.status).to eq(404)
+          expect(json_response[:error]).to eq("Order doesn't exist")
+        end
+      end
+
+      context 'with existing order' do
+        let!(:order) { create(:order, user: user) }
+
+        context 'without line items' do
+          let!(:line_item) { create(:line_item) }
+
+          it 'tries to remove an item and fails' do
+            headers = { 'Authorization' => "Bearer #{token.token}" }
+            delete '/api/v2/storefront/cart/remove_item', params: { line_item_id: line_item.id }, headers: headers
+
+            expect(response.status).to eq(404)
+          end
+        end
+
+        context 'containing line item' do
+          let!(:line_item) { create(:line_item, order: order) }
+
+          it 'removes line item from the cart' do
+            headers = { 'Authorization' => "Bearer #{token.token}" }
+            delete '/api/v2/storefront/cart/remove_item', params: { line_item_id: line_item.id, order_token: guest_token }, headers: headers
```

Comment:
@WitekL let's not use `params: { order_token: guest_token }` - it should be passed as `X-Spree-Order-Token` header, also for the guest checkout flow - we shouldn't pass the `Bearer` header at all